targets/TARGET_RENESAS/TARGET_RZ_A1XX/sleep.c

```diff
@@ -0,0 +1,196 @@
+    {&wk_CPGSTBCR6,  &CPGSTBCR6,  &CPGSTBREQ1, &CPGSTBACK1, CPG_STBCR6_BIT_MSTP66,   CPG_STBREQ1_BIT_STBRQ10}, /* CEU */
+    {&wk_CPGSTBCR6,  &CPGSTBCR8,  &CPGSTBREQ1, &CPGSTBACK1, CPG_STBCR8_BIT_MSTP82,   CPG_STBREQ1_BIT_STBRQ12}, /* EthernetAVB */
+    {&wk_CPGSTBCR7,  &CPGSTBCR7,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR7_BIT_MSTP74,   CPG_STBREQ2_BIT_STBRQ26}, /* Ethernet */
+    {&wk_CPGSTBCR8,  &CPGSTBCR8,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR8_BIT_MSTP83,   CPG_STBREQ2_BIT_STBRQ27}, /* MediaLB */
+    {&wk_CPGSTBCR9,  &CPGSTBCR9,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR9_BIT_MSTP91,   CPG_STBREQ2_BIT_STBRQ25}, /* VDC5_0 */
+#if defined(TARGET_RZA1H)  || defined(TARGET_VK_RZ_A1H)
+    {&wk_CPGSTBCR8,  &CPGSTBCR8,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR8_BIT_MSTP85,   CPG_STBREQ2_BIT_STBRQ21}, /* IMR-LSD */
+    {&wk_CPGSTBCR8,  &CPGSTBCR8,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR8_BIT_MSTP86,   CPG_STBREQ2_BIT_STBRQ22}, /* IMR-LS2_1 */
+    {&wk_CPGSTBCR8,  &CPGSTBCR8,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR8_BIT_MSTP87,   CPG_STBREQ2_BIT_STBRQ23}, /* IMR-LS2_0 */
+    {&wk_CPGSTBCR9,  &CPGSTBCR9,  &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR9_BIT_MSTP90,   CPG_STBREQ2_BIT_STBRQ24}, /* VDC5_1 */
+    {&wk_CPGSTBCR10, &CPGSTBCR10, &CPGSTBREQ2, &CPGSTBACK2, CPG_STBCR10_BIT_MSTP100, CPG_STBREQ2_BIT_STBRQ20}, /* OpenVG */
+#endif
+    {0, 0, 0, 0, 0}  /* None */
+};
+
+static void module_standby_in(void) {
+    volatile uint32_t cnt;
+    volatile uint8_t dummy_8;
+    const module_stanby_t * p_module = &module_stanby[0];
+
+    while (p_module->p_wk_stbcr != 0) {
+        if ((*p_module->p_wk_stbcr & p_module->mstp) == 0) {
+            *p_module->p_stbreq |= p_module->stbrq;
+            dummy_8             = *p_module->p_stbreq;
+            for (cnt = 0; cnt < 1000; cnt++) {      // wait time
+                if ((*p_module->p_stback & p_module->stbrq) != 0) {
+                    break;
+                }
+            }
+            *p_module->p_stbcr |= p_module->mstp;
+            dummy_8            = *p_module->p_stbcr;
+        }
+        p_module++;
+    }
+    (void)dummy_8;
+}
+
+static void module_standby_out(void) {
+    volatile uint32_t cnt;
+    volatile uint8_t dummy_8;
+    const module_stanby_t * p_module = &module_stanby[0];
+
+    while (p_module->p_wk_stbcr != 0) {
+        if ((*p_module->p_wk_stbcr & p_module->mstp) == 0) {
+            *p_module->p_stbreq &= ~(p_module->stbrq);
+            dummy_8             = *p_module->p_stbreq;
+            for (cnt = 0; cnt < 1000; cnt++) {
+                if ((*p_module->p_stback & p_module->stbrq) == 0) {
+                    break;
+                }
+            }
+        }
+        p_module++;
+    }
+    (void)dummy_8;
+}
+
+void hal_sleep(void) {
+    // Transition to Sleep Mode
+    __WFI();
+}
+
+void hal_deepsleep(void) {
+    volatile uint8_t dummy_8;
+
+    core_util_critical_section_enter();
+    wk_CPGSTBCR3  = CPGSTBCR3;
```

Comment:
what are these lines doing? Can you add a commentaas for some you did (line 130 for instance). Could be for entire block. By looking at the lines, I have no understanding of why lines 116 - 127 are doing.

Reply:
This is a process for temporarily saving register values for powerdown the peripheral module.
I added a comment in the code.